Sources/Appcast/SUAppcastDriver.swift

```diff
@@ -8,16 +8,85 @@
 import Foundation
 
 public class SUAppcastDriver {
-    public static func filterSupportedAppcast(_ appcast: SUAppcast, phasedUpdateGroup: NSNumber?, skippedUpdate: Any?, currentDate: Date, hostVersion: String, versionComparator: SUVersionComparison, testOSVersion: Bool, testMinimumAutoupdateVersion: Bool) -> SUAppcast {
-        return appcast
+    public static func itemIsReadyForPhasedRollout(_ item: SUAppcastItem, phasedUpdateGroup: NSNumber?, currentDate: Date, hostVersion: String, versionComparator: SUVersionComparison) -> Bool {
+        if phasedUpdateGroup == nil || item.isCriticalUpdate {
+            return true
+        }
+
+        guard let phasedRolloutInterval = item.phasedRolloutInterval?.doubleValue else {
+            return true
+        }
+
+        guard let itemReleaseDate = item.date else {
+            return true
+        }
+
+        let timeSinceRelease = currentDate.timeIntervalSince(itemReleaseDate)
+        let timeToWaitForGroup = phasedRolloutInterval * Double(phasedUpdateGroup!.uintValue)
+
+        return timeSinceRelease >= timeToWaitForGroup
+    }
+    
+    public static func containsSkippedUpdate(item: SUAppcastItem, skippedUpdate: SPUSkippedUpdate?, hostPassesSkippedMajorVersion: Bool, versionComparator: SUVersionComparison) -> Bool {
+        guard let skippedUpdate = skippedUpdate else {
+            return false
+        }
+
+        let skippedMajorVersion = skippedUpdate.majorVersion
+        let skippedMajorSubreleaseVersion = skippedUpdate.majorSubreleaseVersion
+
+        if !hostPassesSkippedMajorVersion, let skippedMajorVersion = skippedMajorVersion, let minimumAutoupdateVersion = item.minimumAutoupdateVersion, versionComparator.compareVersion(skippedMajorVersion, toVersion: minimumAutoupdateVersion) != .orderedAscending, (item.ignoreSkippedUpgradesBelowVersion == nil || (skippedMajorSubreleaseVersion != nil && versionComparator.compareVersion(skippedMajorSubreleaseVersion!, toVersion: item.ignoreSkippedUpgradesBelowVersion!) != .orderedAscending)) {
```

Comment:
This expression will work, but it can be rewritten using `let` assignments to prevent forced unwrapped statements with `!`:

```swift
(item.ignoreSkippedUpgradesBelowVersion == nil || (skippedMajorSubreleaseVersion != nil && versionComparator.compareVersion(skippedMajorSubreleaseVersion!, toVersion: item.ignoreSkippedUpgradesBelowVersion!)
```

Eg.

```swift
if let skippedMajorSubreleaseVersion,
   let item.ignoreSkippedUpgradesBelowVersion = item.ignoreSkippedUpgradesBelowVersion,
   versionComparator.compareVersion(skippedMajorSubreleaseVersion, toVersion: ignoreSkippedUpgradesBelowVersion) {
   ...
}
```